client/state/selectors/get-param-from-url-or-oauth2-redirect.ts

```diff
@@ -0,0 +1,32 @@
+import { get } from 'lodash';
+import 'calypso/state/route/init';
+import getCurrentQueryArguments from './get-current-query-arguments';
+import type { AppState } from 'calypso/types';
+
+/**
+ * Return the value of the given search query argument.
+ *
+ * Login flow and signup flow use different query arguments to pass the `wccom-from` or `woo-passwordless` values.
+ * Login flow uses the parameter in the url directly, while signup flow uses `oauth2_redirect` to pass the parameter.
+ */
+export const getParamFromUrlOrOauth2Redirect = (
+	state: AppState,
+	paramName: string
+): string | null => {
+	const currentQuery = getCurrentQueryArguments( state );
+	const wccomFrom = get( currentQuery, paramName ) as string | null;
```

Comment:
This is agnostic of `wccom` right? So name should reflect that

Reply:
thanks for catching this! I swear I renamed it but I must be delusional 😆